metricsaggregator/src/main/java/io/dockstore/metricsaggregator/helper/AggregationHelper.java

```diff
@@ -259,6 +261,53 @@ public static Optional<MemoryMetric> getAggregatedMemoryFromExecutions(List<RunE
         return Optional.empty();
     }
 
+    /**
+     * Aggregate Cost metrics from all submissions by calculating the minimum, maximum, and average.
+     * @param allSubmissions
+     * @return
+     */
+    public static Optional<CostMetric> getAggregatedCost(ExecutionsRequestBody allSubmissions) {
+        // Get aggregated Execution Time metrics that were submitted to Dockstore
+        List<CostMetric> costMetrics = allSubmissions.getAggregatedExecutions().stream()
+                .map(Metrics::getCost)
+                .filter(Objects::nonNull)
+                .collect(Collectors.toList());
```

Comment:
You're using this instead of `toList` because you need a mutable list for the next line, right? If so, the `Collectors.toList()` JavaDoc says "There are no guarantees on the type, mutability, serializability, or thread-safety of the {@code List} returned". The current implementation returns an ArrayList, so it's good, but in theory it could break in a future JDK release (but unlikely).

Reply:
Yeah I need a mutable list. Would wrapping it in `ArrayList` be the way to go?

```
List<CostMetric> costMetrics = new ArrayList<>(allSubmissions.getAggregatedExecutions().stream()
                .map(Metrics::getCost)
                .filter(Objects::nonNull)
                .toList());
```

Reply:
That works, you can also:
```
.collect(Collectors.toCollection(ArrayList::new))
```